webapp/src/pages/EditUserPage/EditUserPage.tsx

```diff
@@ -1,12 +1,74 @@
-import { AvatarPageTemplate, EditUserForm } from '../../shared/components';
-import { EDIT_AVATAR_URL, AVATAR_EP_URL } from '../../shared/urls';
+import {
+  AvatarPageTemplate,
+  ButtonVariant,
+  Input,
+  InputVariant,
+} from '../../shared/components';
+import {
+  EDIT_AVATAR_URL,
+  AVATAR_EP_URL,
+  EDIT_USER_PASSWORD_URL,
+  USER_ME_URL,
+} from '../../shared/urls';
 import { useParams } from 'react-router-dom';
 import { useAuth } from '../../shared/hooks/UseAuth';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { ResponseError, UpdateUserDto } from '../../shared/generated';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+import { useState } from 'react';
+import { usersApi } from '../../shared/services/ApiService';
 
 export default function EditUserPage() {
   const { username } = useParams();
   const { authUser, isLoading } = useAuth(username);
+  const { navigate } = useNavigation();
+  const { warn, notify } = useNotificationContext();
+  const initialFormValues: UpdateUserDto = {
+    username: authUser?.username,
+    fullName: authUser?.fullName,
+    email: authUser?.email,
+  };
+  const [formValues, setFormValues] =
+    useState<UpdateUserDto>(initialFormValues);
 
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormValues((previousValues: any) => {
+      return { ...previousValues, [name]: value };
+    });
+  };
+  const handleRequestError = async (error: unknown) => {
+    let errMessage = '';
+    if (error instanceof ResponseError) {
+      if (error.response.status === 400) {
+        errMessage = 'Invalid or missing fields';
+      } else if (error.response.status === 422) {
+        errMessage = 'Username already exists';
+      } else {
+        errMessage = `${error.response.statusText}`;
+      }
+    } else if (error instanceof Error) {
+      errMessage = `${error.message}`;
+    }
+    warn(errMessage);
+  };
+
+  async function updateData() {
+    try {
+      await usersApi.userControllerUpdateCurrentUser({
+        updateUserDto: formValues,
+      });
+      notify('Updated successfully');
+      window.location.href = USER_ME_URL;
```

Comment:
suggestion: Here we could use the `setAuthUser` from `useAuth` and the `navigate` method from `useNavigate`, to prevent the app from loading again..

Reply:
What do you mean with `setAuthUser`? I tried with `navigate` in the begining but navigating to `USER_ME_URL` this ways shows the old page, not updated, so I `used window.location.href`. I don't know if we can use `navigate` and `useEffect` to show the updated values

Reply:
Done